Login Rework: add Tracks

hypest opened the pull request:
During the work for the new Login flows, the Tracks points were ignored in favour of letting the code and features settle and then add the Tracks. This is the PR that re-instates them along with adding a few new ones dedicated to the new Login screens.

aerych commented on WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginActivity.java:
```diff
@@ -267,6 +270,7 @@ public void showMagicLinkSentScreen(String email) {
     @Override
     public void openEmailClient() {
         if (WPActivityUtils.isEmailClientAvailable(this)) {
+            AnalyticsTracker.track(AnalyticsTracker.Stat.LOGIN_MAGIC_LINK_OPEN_EMAIL_CLIENT_VIEWED);
```
This stat should actually be tracking when the `LoginMagicLinkSentFragment` is viewed, rather than when the user attempts to open the email client.  I realize the naming makes that a bit unclear.

hypest replied:
Hey @aerych, I see, I will make the change.

We should probably rename this one to something more clear (like `LOGIN_MAGIC_LINK_SENT_VIEWED`) while it's still early.

We should probably also track the attempt to open the email client. May I suggest introducing a `LOGIN_MAGIC_LINK_OPEN_EMAIL_CLIENT_CLICKED`?

hypest commented:
Thanks for the review @aerych!

> we should bump the _VIEWED stats whenever their respective activity is being viewed.For example, when a user taps the back button to return to a previous activity we want to bump the stat again.

I see. I will look into it.

hypest commented:
Hey @aerych , I've changed the track bumped when the magiclink sent screen is viewed with 4a2830e.

While at it, I added a new track (`LOGIN_MAGIC_LINK_OPEN_EMAIL_CLIENT_CLICKED`) for the click on the open-email-client button. 5022653

Ready for another pass!

I haven't been able to see a problem with the current implementation of _when_ the tracks are bumped for the viewing of the login screens. I tested a few devices and API versions but the tracks are firing both on first and subsequent viewings of the screens (like on Back). Can you share some info on what's the setup you use and see the tracks not-firing on Back?

aerych commented:
> I haven't been able to see a problem with the current implementation of when the tracks are bumped for the viewing of the login screens. I tested a few devices and API versions but the tracks are firing both on first and subsequent viewings of the screens (like on Back). Can you share some info on what's the setup you use and see the tracks not-firing on Back?

Ugh. I'm not sure what happened but when I test again I *do* see the events fired when tapping the back button.  Sorry about that. :( 

> While at it, I added a new track (LOGIN_MAGIC_LINK_OPEN_EMAIL_CLIENT_CLICKED) for the click on the open-email-client button

Nice!!
